File: 3R/10502_ProductOfDigits.java
import java.util.*;

class Main{
	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
        int cases = sc.nextInt();
        
        while(cases-->0){
            int n = sc.nextInt();
            if(n==1) System.out.println(1);
            else{
                StringBuilder sb = new StringBuilder();
                for(int i=9; i>1; i--){
                    while(n%i==0){
                        sb.insert(0, i);
                        n /= i;
                    }
                }
                System.out.println(n==1? sb:"-1");
            }
        }
	}
}

File: 3R/10502_ProductOfDigits.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(){
    int cases;
	cin >> cases;
    while(cases--){
        int n;
        cin >> n;
        if(n==1) cout << 1 << endl;
        else{
            vector<int> vt;
            vector<int>::iterator iter;

            for(int i=9; i>1; i--){
                while(n%i==0){
                    vt.insert(vt.begin(), i);
                    n/=i;
                }
            }

            if(n==1){
                for(iter = vt.begin(); iter!=vt.end(); iter++){
                    cout << *iter;
                }
                cout << endl;
            }else{
                cout << "-1" << endl;
            }
        }
    }
    return 0;
}